src/sentry/integrations/github/repository.py

```diff
@@ -82,35 +68,75 @@ def compare_commits(self, repo, start_sha, end_sha):
-            if integration_id is None:
-                raise NotImplementedError('GitHub apps requires an integration id to fetch commits')
-            integration = Integration.objects.get(id=integration_id)
-            installation = integration.get_installation(repo.organization_id)
-            client = installation.get_client()
-
-            # use config name because that is kept in sync via webhooks
-            name = repo.config['name']
-            try:
-                res = client.get_pr_commits(name, number)
-            except Exception as e:
-                installation.raise_error(e)
-            else:
-                return self._format_commits(repo, res)
+                return self._format_commits(client, name, res['commits'])
+        except Exception as e:
+            installation.raise_error(e)
+
+    def _format_commits(self, client, repo_name, commit_list):
+        """Convert GitHub commits into our internal format
+
+        For each commit in the list we have to fetch patch data, as the
+        compare API gives us all of the files changed in the commit
+        range but not which files changed in each commit. Without this
+        we cannot know which specific commit changed a given file.
+
+        See sentry.models.Release.set_commits
+        """
+        return [
+            {
+                'id': c['sha'],
+                'repository': repo_name,
+                'author_email': c['commit']['author'].get('email'),
+                'author_name': c['commit']['author'].get('name'),
+                'message': c['commit']['message'],
+                'timestamp': dateutil.parser.parse(
+                    c['commit']['author'].get('date'),
+                ).astimezone(timezone.utc) if c['commit']['author'].get('date') else None,
+                'patch_set': self._get_patchset(client, repo_name, c['sha'])
+            } for c in commit_list
+        ]
+
+    def _get_patchset(self, client, repo_name, sha):
+        """Get the modified files for a commit
+        """
+        commit = client.get_commit(repo_name, sha)
+        return self._transform_patchset(commit['files'])
+
+    def _transform_patchset(self, diff):
+        """Convert the patch data from GitHub into our internal format
+
+        See sentry.models.Release.set_commits
+        """
+        changes = []
+        for change in diff:
+            if change['status'] == 'modified':
+                changes.append({
+                    'path': change['filename'],
+                    'type': 'M',
+                })
+            if change['status'] == 'added':
+                changes.append({
+                    'path': change['filename'],
+                    'type': 'A',
+                })
+            if change['status'] == 'removed':
+                changes.append({
+                    'path': change['filename'],
+                    'type': 'D',
+                })
+            if change['status'] == 'renamed':
```

Comment:
We don't seem to have `renamed` in `integrations/github/webhook/` should that be added there too?

Reply:
Probably. I'll double check that `renamed` comes through on the pull request payload as GitHub isn't always 100% accurate.

Reply:
I created APP-697 to follow up on this.